Sources/TuistGenerator/Writers/XcodeProjWriter.swift

```diff
@@ -134,6 +134,10 @@ public final class XcodeProjWriter: XcodeProjWriting {
     ) throws {
         let settingsPath = WorkspaceSettingsDescriptor.xcsettingsFilePath(relativeToWorkspace: xccontainerPath)
 
+        let parentFolder = settingsPath.removingLastComponent()
+        if !FileHandler.shared.exists(parentFolder) {
+            try FileHandler.shared.createFolder(parentFolder)
+        }
```

Comment:
Now we are always creating the Xcode settings because TextSettings is no longer nillable (but its properties are), and in tuist edit the `*.xcworkspace/xcshareddata` folder was missing, hence the write of the file inside it was failing